libqtile/widget/generic_poll_text.py

```diff
@@ -92,3 +93,36 @@ def poll(self):
             text = "Can't parse"
 
         return text
+
+
+class GenPollCommand(base.ThreadPoolText):
+    """A generic text widget to display output from scripts or shell commands"""
+
+    defaults = [
+        ("update_interval", 60, "update time in seconds"),
+        ("cmd", None, "command line as a string or list of arguments to execute"),
+        ("shell", False, "run command through shell to enable piping and shell expansion"),
+    ]
+
+    def __init__(self, **config):
+        base.ThreadPoolText.__init__(self, "", **config)
+        self.add_defaults(GenPollCommand.defaults)
+
+    def _configure(self, qtile, bar):
+        base.ThreadPoolText._configure(self, qtile, bar)
+
+        self.add_callbacks({
+            "Button1": self.force_update,
+        })
+
+    def poll(self):
+        return self.run_process().stdout.strip()
+
+    def run_process(self):
+        self.completed_process = subprocess.run(
+            self.cmd,
+            capture_output=True,
+            text=True,
+            shell=self.shell,
+        )
+        return self.completed_process
```

Comment:
```suggestion
    def poll(self):
        process = subprocess.run(
            self.cmd,
            capture_output=True,
            text=True,
            shell=self.shell,
        )
        return process.stdout.strip()
```

There's no need for `run_process` to be a separate method. I'd keep it all in `poll`.